helix-core/src/syntax.rs

```diff
@@ -23,6 +25,34 @@ use std::{
 use once_cell::sync::{Lazy, OnceCell};
 use serde::{Deserialize, Serialize};
 
+#[cfg(unix)]
+pub const DYLIB_EXTENSION: &str = "so";
+
+#[cfg(windows)]
+pub const DYLIB_EXTENSION: &str = "dll";
+
+fn replace_dashes_with_underscores(name: &str) -> String {
+    name.replace('-', "_")
+}
+
+pub fn get_language(runtime_path: &std::path::Path, name: &str) -> Result<Language> {
+    let name = name.to_ascii_lowercase();
+    let mut library_path = runtime_path.join("grammars").join(&name);
+    library_path.set_extension(DYLIB_EXTENSION);
+
+    let library = unsafe { Library::new(&library_path) }
+        .with_context(|| format!("Error opening dynamic library {:?}", &library_path))?;
+    let language_fn_name = format!("tree_sitter_{}", replace_dashes_with_underscores(&name));
+    let language = unsafe {
+        let language_fn: Symbol<unsafe extern "C" fn() -> Language> = library
+            .get(language_fn_name.as_bytes())
+            .with_context(|| format!("Failed to load symbol {}", language_fn_name))?;
+        language_fn()
+    };
+    std::mem::forget(library);
+    Ok(language)
+}
+
```

Comment:
Hmm yeah most of what's in here seems like it could be moved out of core. Maybe it would make sense to keep the `helix-syntax` crate and make the grammar fetching/building stuff a module within that crate?

Reply:
I think for this to work well we'd need to move config out of core. Although a crate just for config seems like maybe too many crates. I'm curious to hear what you think @archseer?

Reply:
I would move the new grammar config into `helix-loader`, then import it into `helix-core`. In other words, `helix-core` should depend on the loader. Let's keep the rest of syntax.rs here because it's specific to highlighting.